packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -80,16 +80,49 @@ export const useStoreSwitcher = props => {
         // Change store view code and currency to be used in Apollo link request headers
         storeCode => {
             // Do nothing when store view is not present in available stores
-            if (!availableStores.get(storeCode)) return;
+            if (!availableStores.has(storeCode)) return;
 
             storage.setItem('store_view_code', storeCode);
             storage.setItem(
                 'store_view_currency',
                 availableStores.get(storeCode).currency
             );
 
-            // Refresh the page to re-trigger the queries once code/currency are saved in local storage.
-            history.go(0);
+            // Handle updating the URL if the store code should be present
+            if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+                const pathName = window.location.pathname;
+
+                // Check to see if we're on a page outside of the homepage
+                if (pathName !== '' && pathName !== '/') {
+                    const pathNameSplit = pathName.split('/');
+                    const pathStoreCode = pathNameSplit[1];
+
+                    // Is the current store code in the URL? If so replace it with the new one
+                    if (
+                        availableStores.has(pathStoreCode) &&
+                        availableStores.get(pathStoreCode).isCurrent
+                    ) {
+                        window.location.assign(
+                            window.location.pathname.replace(
+                                `/${pathStoreCode}`,
+                                `/${storeCode}`
+                            )
+                        );
+                    } else {
+                        // If there is no store code present, include it and reload
+                        window.location.assign(
+                            `/${storeCode}${
+                                pathName && pathName !== '/' ? pathName : ''
+                            }`
+                        );
+                    }
```

Comment:
I feel like this could be simplified. Strip out any existing store code in the url and add in the selected one plus all the rest of the path and params.

Reply:
I see now that this was all necessary to ensure we don't mistake a path for a store code. ie

`https://example.com/search.html` where `search.html` would be `pathName` but not a valid store code. In that case we need to insert the store code before the path.